### PR TITLE
feat (docs): error handling for generateObject

### PR DESCRIPTION
**Pull request by lgrammel**

### CODE REVIEW

File: content/docs/03-ai-sdk-core/10-generating-structured-data.mdx [hunk 1]
```diff
@@ -111,3 +111,44 @@ const result = await generateObject({
   prompt: 'Generate a fake user profile for testing.',
 });
 ```
+
+## Error Handling
+
+When you use `generateObject`, errors are thrown when the model fails to generate property JSON (`JSONParseError`)
```
Comment (nicoalbanese) [inline content/docs/03-ai-sdk-core/10-generating-structured-data.mdx:117]: "proper JSON"?

File: content/docs/03-ai-sdk-core/10-generating-structured-data.mdx [hunk 2]
```diff
@@ -111,3 +111,44 @@ const result = await generateObject({
...
+Both error types contain additional information, e.g. the generated text or the invalid value.
+
+You can use this to e.g. design a function that safely process the result object and also returns values in error cases:
+
+```ts
+import { openai } from '@ai-sdk/openai';
+import { JSONParseError, TypeValidationError, generateObject } from 'ai';
+import { z } from 'zod';
+
+async function example<T>(
```
Comment (nicoalbanese) [inline content/docs/03-ai-sdk-core/10-generating-structured-data.mdx:128]: should we name this something more in line with what it's doing?

**Review by nicoalbanese — APPROVED** [review]

File: content/docs/03-ai-sdk-core/10-generating-structured-data.mdx (hunk 1 above)
Comment (lgrammel) [reply content/docs/03-ai-sdk-core/10-generating-structured-data.mdx:117]: yes thank you

File: content/docs/03-ai-sdk-core/10-generating-structured-data.mdx (hunk 2 above)
Comment (lgrammel) [reply content/docs/03-ai-sdk-core/10-generating-structured-data.mdx:128]: good idea

**Pull request merged**